File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.py
# Last updated: 9/16/2025, 12:52:23 AM
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        word = []
        for i in range(0,max(len(word1),len(word2))):
            if i < len(word1):
                word.append(word1[i])
            if i < len(word2):
                word.append(word2[i])
        return "".join(word)
        

File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.cpp
// Last updated: 2/13/2026, 12:56:18 PM
class Solution {
public:
    string mergeAlternately(string word1, string word2) 
    {
        int i = 0, j = 0;
        string w;
        while( i <word1.size()&& j <word2.size())
        {
            w += word1[i];
            w += word2[j];
            i++; j++;
        }
        while(i<word1.size())
        {
            w += word1[i];
            i++;
        }
        
        while(j<word2.size())
        {
            w += word2[j];
            j++;
        }

        return w;
    }
};